subsys/net/lib/shell/iface.c

```diff
@@ -257,6 +262,39 @@ static void iface_cb(struct net_if *iface, void *user_data)
 			   orig_iface);
 		}
 	}
+
+	if (IS_ENABLED(CONFIG_NET_VPN) &&
+	    net_if_l2(iface) == &NET_L2_GET_NAME(VIRTUAL)) {
+		if (net_virtual_get_iface_capabilities(iface) & VIRTUAL_INTERFACE_VPN) {
+			struct virtual_interface_req_params vparams = { 0 };
+			char public_key[NET_VIRTUAL_MAX_PUBLIC_KEY_LEN * 2];
+			size_t olen;
+
+			ret = net_mgmt(NET_REQUEST_VIRTUAL_INTERFACE_GET_PUBLIC_KEY,
+				       iface, &vparams, sizeof(vparams));
+			if (ret < 0) {
+				PR_WARNING("Cannot get VPN public key (%d)\n", ret);
+			} else {
+				int count_zeros = 0;
+
+				for (int i = 0; i < NET_VIRTUAL_MAX_PUBLIC_KEY_LEN; i++) {
+					if (vparams.public_key.data[i] == 0) {
+						count_zeros++;
+					}
+				}
+
+				if (count_zeros == NET_VIRTUAL_MAX_PUBLIC_KEY_LEN) {
+					PR("Public key: <not set>\n");
+				} else {
+					(void)base64_encode(public_key, sizeof(public_key),
+							    &olen, vparams.public_key.data,
+							    vparams.public_key.len);
+
+					PR("Public key: %s\n", public_key);
+				}
```

Comment:
Minor optimization, we don't need to count all zeros IMO, finding a single non-zero value is sufficient?

```c
				bool all_zeros = true;

				for (int i = 0; all_zeros && i < NET_VIRTUAL_MAX_PUBLIC_KEY_LEN; i++) {
					all_zeros = vparams.public_key.data[i] == 0;
				}

				if (all_zeros) {
					PR("Public key: <not set>\n");
				} else {
					(void)base64_encode(public_key, sizeof(public_key),
							    &olen, vparams.public_key.data,
							    vparams.public_key.len);

					PR("Public key: %s\n", public_key);
				}
```

Reply:
True, will fix that.